contracts/RelayHub.sol

```diff
@@ -238,45 +238,36 @@ contract RelayHub is IRelayHub {
     )
     public
     view
-    returns (uint256 status, bytes memory recipientContext)
+    returns (bool success, string memory returnValue)
     {
-        {
-            // Verify the sender's request: signature and nonce.
-            ITrustedForwarder forwarder = ITrustedForwarder(BaseRelayRecipient(relayRequest.target).getTrustedForwarder());
-            (bool success, bytes memory ret) = address(forwarder).staticcall(abi.encodeWithSelector(
-                    forwarder.verify.selector,
-                    relayRequest, signature
-                ));
-            if (!success) {
-                ret = bytes(GsnUtils.getError(ret));
-                //temporary check (until we get rid of the status codes, and rely only on description)
-                if (ret[0] == "n")
-                    return (uint256(CanRelayStatus.WrongNonce), ret);
-                else
-                    return (uint256(CanRelayStatus.WrongSignature), ret);
-            }
+        bytes memory ret;
+        (success, ret) = relayRequest.target.staticcall(abi.encodeWithSelector(
+            BaseRelayRecipient(relayRequest.target).getTrustedForwarder.selector
+        ));
+        if (!success || ret.length != 32) {
+            return (false, "getTrustedForwarder failed");
+        }
+        // Verify the sender's request: signature and nonce.
+        ITrustedForwarder forwarder = ITrustedForwarder(abi.decode(ret, (address)));
+        (success, ret) = address(forwarder).staticcall(abi.encodeWithSelector(
+                forwarder.verify.selector,
+                relayRequest, signature
+            ));
+        if (!success) {
+            return (false, GsnUtils.getError(ret));
         }
 
         bytes memory encodedTx = abi.encodeWithSelector(IPaymaster(address(0)).acceptRelayedCall.selector,
             relayRequest, approvalData, maxPossibleGas
         );
 
-        (bool success, bytes memory returndata) =
+        (success, ret) =
         relayRequest.relayData.paymaster.staticcall.gas(acceptRelayedCallGasLimit)(encodedTx);
 
         if (!success) {
-            return (uint256(CanRelayStatus.AcceptRelayedCallReverted), "AcceptRelayedCallReverted");
-        } else {
-            (status, recipientContext) = abi.decode(returndata, (uint256, bytes));
-
-            // This can be either CanRelayStatus.OK or a custom error code
-            if ((status == 0) || (status > 10)) {
-                return (status, recipientContext);
-            } else {
-                // Error codes [1-10] are reserved to RelayHub
-                return (uint256(CanRelayStatus.InvalidRecipientStatusCode), "InvalidRecipientStatusCode");
-            }
+            return (false, GsnUtils.getError(ret));
         }
+        returnValue = abi.decode(ret, (string));
```

Comment:
No. the context is the return value, without decode.
The reason I made canRelay return `string` and not `bytes`, is to ease the client parse (which cares nothing about the context, only about the error string).
on-chain is simply `string(x)` or `bytes(y)` to convert between them.